tools/build_defs/repo/utils.bzl

```diff
@@ -112,6 +112,23 @@ def download_remote_files(ctx, auth = None):
     # Wait until the requests are done
     return {path: token.wait() for path, token in pending.items()}
 
+def symlink_files(ctx):
+    # type: (repository_ctx) -> None
+    """Utility function for symlinking local files.
+
+    This is intended to be used in the implementation function of a repository rule. It assumes the
+    parameter `files` is present in `ctx.attr`.
+
+    Args:
+      ctx: The repository context of the repository rule calling this utility
+        function.
+    """
+    for path, label in ctx.attr.files.items():
+        src_path = ctx.path(label)
+        if not src_path.exists:
+            fail("Input %s does not exist" % label)
+        ctx.symlink(src_path, path)
```

Comment:
What happens if `path` already exists? I think we should clarify this in the doc.